Formula/bun.rb

```diff
@@ -0,0 +1,65 @@
+class Bun < Formula
+  desc "Fast all-in-one JavaScript runtime"
+  homepage "https://bun.sh"
+  # pull from git tag to get submodules
+  url "https://github.com/oven-sh/bun.git",
+    tag:      "bun-v0.1.2",
+    revision: "1ee94d5bd261b4cfad7798f0e119499f351b93f3"
+  license "MIT"
+  head "https://github.com/oven-sh/bun.git", branch: "main"
+
+  livecheck do
+    url :stable
+    regex(/bun-v?(\d+(?:\.\d+)+)/i)
+  end
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "cmake" => :build
+  depends_on "esbuild" => :build
+  depends_on "gnu-sed" => :build
+  depends_on "go" => :build
+  depends_on "libtool" => :build
+  depends_on "llvm@13" => :build
+  depends_on "ninja" => :build
+  depends_on "node" => :build
+  depends_on "rust" => :build
+
+  # Fails to build with latest zig release
+  # Commit from latest fork: https://github.com/Jarred-Sumner/zig
+  resource "zig" do
+    url "https://github.com/ziglang/zig/archive/0ea51f7f494cd84a48fd997b60196d6c4254ccac.tar.gz"
+    sha256 "7a6cf86adb7e7631f16a9ed2b8fcf461ed5cb387cc9b30f38bf182fa235b446f"
+  end
```

Comment:
https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae

This will need to be contributed upstream before we can use it.

Reply:
Note that this uses `ziglang/zig`, so there is nothing to upstream. The `Jarred-Sumner/zig` fork only seemingly serves as a pointer to the latest compatible zig commit

Speaking of vendoring, there are quiet a few vendored deps that we'll want to replace: https://github.com/oven-sh/bun/blob/main/.gitmodules